client/components/domains/registrant-extra-info/fr-form.jsx

```diff
@@ -116,13 +125,19 @@ class RegistrantExtraInfoFrForm extends React.PureComponent {
 						{ translate( 'VAT Number' ) }
 						{ this.renderOptional() }
 					</FormLabel>
-					<FormTextInput
-						id="registrantVatId"
+					<Input
+						name="registrantVatId"
 						value={ registrantVatId }
 						autoCapitalize="off"
 						autoComplete="off"
 						autoCorrect="off"
 						placeholder={ translate( 'ex. XX123456789' ) }
+						isError={ Boolean( validationErrors.registrantVatId ) }
```

Comment:
There's `formState` for multiple field forms. You can check one https://github.com/Automattic/wp-calypso/blob/8db70cf3a584cd679ed884f25337da3b2140a397/client/my-sites/domains/domain-management/email-forwarding/email-forwarding-add-new.jsx

Reply:
We worked with the `formState` controller on the `contactDetailsForm`, and did some pretty horrible things to be able to use it there, but I don't think it's a particularly good fit here.

Between `Redux` and client-side validation, a lot of it's functionality is unnecessary. Loading, submitting, and handling changes go straight in and out of props, and the validation seems fast enough to just do it(caveat: no performance testing), so there's no need for extra callbacks and tracking `isPendingValidation` et al.

It does mean I'm reinventing the wheel a bit with initializing default values, but unless you strongly disagree, I'd prefer to stick with this for now.